src/store/middlewares/notification.js

```diff
@@ -0,0 +1,23 @@
+import actionTypes from '../../constants/actions';
+import Notification from '../../utils/notification';
+
+const notificationMiddleware = (store) => {
+  const notify = Notification.init();
+  return next => (action) => {
+    next(action);
+
+    const { account } = store.getState();
```

Comment:
So I figured out that the reason it wasn't showing the notification is actually here. Lines 7 and 9 need to be swapped. 

Currently, `next(action);` goes synchronously through all other middlewares and reducers, then `{ account }` gets the updated value of the account, so there is never any difference in balance.